<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EWSMainWindowForm</class>
 <widget class="QMainWindow" name="EWSMainWindowForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Visualize Physics: Wave</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>100</horstretch>
     <verstretch>100</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="ews::app::widget::QOSGWidget" name="renderer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
        <horstretch>100</horstretch>
        <verstretch>100</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>10</width>
        <height>10</height>
       </size>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="margin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>25</height>
    </rect>
   </property>
   <property name="nativeMenuBar" stdset="0">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionReset"/>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuItems">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Items</string>
    </property>
    <addaction name="actionAddBarrier"/>
    <addaction name="actionRemoveBarrier"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionProjectWebsite"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuItems"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <property name="iconSize">
    <size>
     <width>22</width>
     <height>22</height>
    </size>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="separator"/>
   <addaction name="actionReset"/>
  </widget>
  <widget class="QDockWidget" name="barrierControls">
   <property name="minimumSize">
    <size>
     <width>56</width>
     <height>56</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="margin">
      <number>12</number>
     </property>
     <item>
      <widget class="ews::app::widget::BarrierEditor" name="barrierEditor" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>100</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>1</width>
         <height>1</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_5">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>:/images/images/file-open-green.png</normaloff>:/images/images/file-open-green.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="icon">
    <iconset>
     <normaloff>:/images/images/view-refresh.png</normaloff>:/images/images/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="toolTip">
    <string>Reset simulation</string>
   </property>
   <property name="shortcut">
    <string>Meta+Shift+R</string>
   </property>
  </action>
  <action name="actionAddBarrier">
   <property name="icon">
    <iconset>
     <normaloff>:/images/images/list-add.png</normaloff>:/images/images/list-add.png</iconset>
   </property>
   <property name="text">
    <string>Add Barrier</string>
   </property>
   <property name="toolTip">
    <string>Add barrier</string>
   </property>
   <property name="shortcut">
    <string>Meta+B</string>
   </property>
  </action>
  <action name="actionRemoveBarrier">
   <property name="icon">
    <iconset>
     <normaloff>:/images/images/list-remove.png</normaloff>:/images/images/list-remove.png</iconset>
   </property>
   <property name="text">
    <string>Remove Barrier</string>
   </property>
   <property name="toolTip">
    <string>Remove selected barrier</string>
   </property>
   <property name="shortcut">
    <string>Meta+Shift+B</string>
   </property>
  </action>
  <action name="actionToggleDripSource1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toggle Drip Source 1</string>
   </property>
   <property name="toolTip">
    <string>Toggle Drip Source 1</string>
   </property>
  </action>
  <action name="actionToggleDripSource2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toggle Drip Source 2</string>
   </property>
   <property name="toolTip">
    <string>Toggle Drip Source 2</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionProjectWebsite">
   <property name="text">
    <string>Project Website</string>
   </property>
  </action>
  <action name="actionGraphicsSystemInfo">
   <property name="text">
    <string>Graphics System Info...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Preferences...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
   <property name="menuRole">
    <enum>QAction::PreferencesRole</enum>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ews::app::widget::QOSGWidget</class>
   <extends>QWidget</extends>
   <header>QOSGWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ews::app::widget::BarrierEditor</class>
   <extends>QWidget</extends>
   <header>BarrierEditor.h</header>
   <slots>
    <slot>addBarrier()</slot>
    <slot>removeBarrier()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddBarrier</sender>
   <signal>triggered()</signal>
   <receiver>barrierEditor</receiver>
   <slot>addBarrier()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1011</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveBarrier</sender>
   <signal>triggered()</signal>
   <receiver>barrierEditor</receiver>
   <slot>removeBarrier()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1011</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionProjectWebsite</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>projectWebsite()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>581</x>
     <y>447</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>EWSMainWindowForm</receiver>
   <slot>preferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>581</x>
     <y>447</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>reset()</slot>
  <slot>about()</slot>
  <slot>projectWebsite()</slot>
  <slot>preferences()</slot>
 </slots>
</ui>
